src/Widgets/Popovers/AvatarPopover.vala

```diff
@@ -84,18 +84,17 @@ namespace SwitchboardPlugUserAccounts.Widgets {
             file_dialog.update_preview.connect (() => {
                 string uri = file_dialog.get_preview_uri ();
                 if (uri != null && uri.has_prefix ("file://")) {
-                    preview_area.set_from_gicon_async.begin (
```

Comment:
It looks like `Gdk.Pixbuf` does implement `loadable_icon` so we should be able to still use `set_from_gicon_async` here I think https://valadoc.org/gdk-pixbuf-2.0/Gdk.Pixbuf.html

Reply:
The problem I encountered was that using `set_from_gicon_async` on an `AsyncImage` would (1) not apply the embedded orientation and (2) the resulting AsyncImage did not expose a `.pixbuf` member that I could apply the transformation to. So that was why I resorted to creating a plain `Gtk.Image` from a `Pixbuf` that was loaded from a file.

(I also noticed that `AsyncImage` was [removed from](https://github.com/elementary/granite/commit/34ec922ca720f4effe0b6e35d77574193d3af7c3) Granite last year, so I thought it wouldn't be a bad idea to stop using it here).

I’m not very familiar with these APIs, and I’m afraid I don’t fully understand your suggestion.

Reply:
Oh yup, I missed that. Ignore me then :) Thanks for explaining!